CMakeLists.txt

```diff
@@ -235,13 +235,21 @@ if(NOT nlohmann_json_FOUND)
       OFF
       CACHE INTERNAL "")
   set(JSON_Install
-      OFF
+      ON
       CACHE INTERNAL "")
-  # This option allows to link nlohmann_json::nlohmann_json target
-  add_subdirectory(${PROJECT_SOURCE_DIR}/third_party/nlohmann-json)
-  # This option allows to add header to include directories
-  include_directories(
-    ${PROJECT_SOURCE_DIR}/third_party/nlohmann-json/single_include)
+  if(EXISTS ${PROJECT_SOURCE_DIR}/third_party/nlohmann-json)
+    # This option allows to link nlohmann_json::nlohmann_json target
+    add_subdirectory(${PROJECT_SOURCE_DIR}/third_party/nlohmann-json)
+    # This option allows to add header to include directories
+    include_directories(
+      ${PROJECT_SOURCE_DIR}/third_party/nlohmann-json/single_include)
+  else()
+    message(
+      FATAL_ERROR
+        "\nnlohmann_json package was not found. Please either provide it manually or clone with submodules. "
+        "To initialize, fetch and checkout any nested submodules, you can use the following command:\n"
+        "git submodule update --init --recursive")
+  endif()
```

Comment:
while we are here, can we also add the change discussed earlier : https://github.com/open-telemetry/opentelemetry-cpp/pull/865#discussion_r654111208 ?
We do have an issue also for that now: https://github.com/open-telemetry/opentelemetry-cpp/issues/869, and I don't think we should need to checkout submodules for api only build.

Reply:
I'll address this for `nlohmann` for API-only build. But I anticipate other issues (both related to `nlohmann` json and unrelated to it) with building SDK from source tarball that does not cover recursive dependencies of SDK.

Reply:
@lalitb - please take a look at latest iteration. This should now pass the API-only build, fixing #869